webflow-hacks/cms-hacks/58/58.ts

```diff
@@ -0,0 +1,66 @@
+// inject the youtube api script
+const tag = document.createElement('script');
+tag.src = 'https://www.youtube.com/iframe_api';
+document.head.insertAdjacentElement('afterbegin', tag);
+
+/**
+ * default function called by the youtube api
+ */
+function onYouTubeIframeAPIReady() {
+  const IFRAME_SELECTOR = '[fs-hacks-element="iframe"]';
+  const iFrameContainer = document.querySelector(IFRAME_SELECTOR);
+  if (!iFrameContainer) return;
+
+  const iFrames = iFrameContainer.querySelectorAll('iframe');
+  iFrames.forEach((iFrame) => {
+    // set iframe properties
+    const { src } = iFrame;
+    if (!src) return;
+
+    const newSrc = new URL(src);
+    if (!newSrc.origin.includes('youtube')) return;
+
+    newSrc.searchParams.append('enablejsapi', '1');
+    iFrame.src = newSrc.toString();
+
+    createPlayer(iFrame);
+  });
+}
+
+/**
+ * Set up the player, add listeners, and play the video.
+ * @param iframe The iframe
+ */
+function createPlayer(iframe: HTMLIFrameElement) {
+  // initialize YT.player object
+  const player = new YT.Player(iframe, {
+    events: {
+      onReady: onPlayerReady,
+    },
+  });
+
+  function onPlayerReady() {
+    const TIMESTAMP_LINK_SELECTOR = '[fs-hacks-element="timestamp"]';
+    const timestampLinks = document.querySelectorAll<HTMLAnchorElement>(TIMESTAMP_LINK_SELECTOR);
+    timestampLinks.forEach((timestampLink) => {
```

Comment:
The last problem I see is that this will not work correctly in the case where there is more than one YouTube player on the page.
Imagine this use case:
- There are two players on the page.
- Each player has it's own timestamps buttons.

You only want to instantiate the timestamps that belong to each player.